chsdi/static/doc/examples/ol3_lv95.js

```diff
@@ -0,0 +1,112 @@
+var RESOLUTIONS = [
+    4000, 3750, 3500, 3250, 3000, 2750, 2500, 2250, 2000, 1750, 1500, 1250,
+    1000, 750, 650, 500, 250, 100, 50, 20, 10, 5, 2.5, 2, 1.5, 1, 0.5
+];
+
+var extent = [2420000, 130000, 2900000, 1350000];
+
+function qualifyURL(url) {
+    var a = document.createElement('a');
+    a.href = url;
+    return a.cloneNode(false).href;
+}
+
+var matrixIds = [];
+for (var i = 0; i < RESOLUTIONS.length; i++) {
+    matrixIds.push(i);
+}
+
+
+var wmtsSource = function(layer, options) {
+    var resolutions = options.resolutions ? options.resolutions : RESOLUTIONS;
+    var tileGrid = new ol.tilegrid.WMTS({
+        origin: [extent[0], extent[3]],
+        resolutions: resolutions,
+        matrixIds: matrixIds
+    });
+    var extension = options.format || 'png';
+    var timestamp = options['timestamps'][0];
+    return new ol.source.WMTS(({
+        attributions: [new ol.Attribution({
+            html: '<a target="new" href="http://www.swisstopo.admin.ch/' +
+                'internet/swisstopo/en/home.html">swisstopo</a>'
+        })],
+        url: (qualifyURL('..') + '1.0.0/{Layer}/default/' +
+            timestamp + '/2056/' +
+            '{TileMatrix}/{TileCol}/{TileRow}.').replace('http:', location.protocol) + extension,
+        tileGrid: tileGrid,
+        layer: options['serverLayerName'] ? options['serverLayerName'] : layer,
+        requestEncoding: 'REST'
+    }));
+};
+
+// See http://api3.geo.admin.ch/rest/services/api/MapServer/layersConfig
```

Comment:
Cons is that it something that we have to maintain for eternity.
@loicgasser: is it based on the ESRI Spec ?

On Tue, Mar 3, 2015 at 7:44 AM, davidoesch <notifications@github.com> wrote:

> In chsdi/static/doc/examples/ol3_lv95.js
> <https://github.com/geoadmin/mf-chsdi3/pull/1270#discussion_r25666163>:
>
> > +    var timestamp = options['timestamps'][0];
> > +    return new ol.source.WMTS(({
> > +        attributions: [new ol.Attribution({
> > +            html: '<a target="new" href="http://www.swisstopo.admin.ch/' +
> > +                'internet/swisstopo/en/home.html">swisstopo</a>'
> > +        })],
> > +        url: (qualifyURL('..') + '1.0.0/{Layer}/default/' +
> > +            timestamp + '/2056/' +
> > +            '{TileMatrix}/{TileCol}/{TileRow}.').replace('http:', location.protocol) + extension,
> > +        tileGrid: tileGrid,
> > +        layer: options['serverLayerName'] ? options['serverLayerName'] : layer,
> > +        requestEncoding: 'REST'
> > +    }));
> > +};
> > +
> > +// See http://api3.geo.admin.ch/rest/services/api/MapServer/layersConfig
>
> Makes sense- tip has already given. What are the cons? @mariokeusen
> <https://github.com/mariokeusen>  can you then inform bovier?
> … <#14bde61336c2d5b6_>
> On Tue, Mar 3, 2015 at 7:39 AM, Cédric Moullet <notifications@github.com>
> wrote: > + var timestamp = options['timestamps'][0]; > + return new
> ol.source.WMTS(({ > + attributions: [new ol.Attribution({ > + html: '<a
> target="new" href="http://www.swisstopo.admin.ch/' + > +
> 'internet/swisstopo/en/home.html">swisstopo</a>' > + })], > + url:
> (qualifyURL('..') + '1.0.0/{Layer}/default/' + > + timestamp + '/2056/' + >
> + '{TileMatrix}/{TileCol}/{TileRow}.').replace('http:', location.protocol)
> + extension, > + tileGrid: tileGrid, > + layer: options['serverLayerName']
> ? options['serverLayerName'] : layer, > + requestEncoding: 'REST' > + }));
> > +}; > + > +// See
> http://api3.geo.admin.ch/rest/services/api/MapServer/layersConfig
> @davidoesch <https://github.com/davidoesch> ? --- Reply to this email
> directly or view it on GitHub:
> https://github.com/geoadmin/mf-chsdi3/pull/1270/files#r25666031
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/geoadmin/mf-chsdi3/pull/1270/files#r25666163>.
>



-- 
Twitter: http://twitter.com/cedricmoullet
Linked In: http://www.linkedin.com/in/cedricmoullet

<http://twitter.com/cedricmoullet>